client-src/elements/chromedash-survey-questions.ts

```diff
@@ -0,0 +1,177 @@
+    const value: boolean = this.gate.survey_answers[name];
+    return html`
+      <li class="question">
+        <sl-checkbox
+          name=${name}
+          ?checked=${value}
+          @sl-change=${e => this.handleFieldChange(name, e.target?.checked)}
+        ></sl-checkbox>
+        ${desc}
+      </li>
+    `;
+  }
+
+  renderStringField(name: string, desc: string): TemplateResult {
+    const value: string = this.gate.survey_answers[name];
+    return html`
+      <li class="question">
+        ${desc}
+        <sl-input
+          name="${name}"
+          size="small"
+          value=${value}
+          @sl-change=${e => this.handleFieldChange(name, e.target?.value)}
+        ></sl-input>
+      </li>
+    `;
+  }
+
+  renderPrivacyForm(): TemplateResult {
+    return html`
+      <div id="questionnaire">
+        <ol>
+          ${this.renderBooleanField(
+            'is_language_polyfill',
+            `This is a new JS language construct that has already
+         been polyfillable.`
+          )}
+          ${this.renderBooleanField(
+            'is_api_polyfill',
+            `This is a new API that ergonomically provides a function
+         that was already polyfillable under the same conditions.
+         By "same conditions" we mean, for example, that if a
+         polyfill was only possible when the user has granted a
+         certain permission, the API respects the same permission.`
+          )}
+          ${this.renderBooleanField(
+            'is_same_origin_css',
+            `This is a CSS addition or change such that the style only
+         depends on same-origin information and does NOT on user data.
+           CSS changes are usually benign, however: if the style relies
+         on iframes or subresources, it could reveal cross-origin
+           information.  If the style depends on user data, such as
+          browsing history (like :visited), cookies, or user input
+        (like hidden=until-found), the style could be used by the
+        website to read this data.`
+          )}
+          ${this.renderStringField(
+            'launch_or_contact',
+            `If there is a Google-internal launch entry filed for this
+         exact same issue, enter its URL here. Or, if this has previously
+        been discussed with someone on the privacy team, enter their
+        email address. Or, enter "None".`
+          )}
+        </ol>
+      </div>
+    `;
+  }
+
+  renderQuestionnaire(): TemplateResult {
+    if (
+      this.gate.gate_type === GATE_TYPES.PRIVACY_ORIGIN_TRIAL ||
+      this.gate.gate_type === GATE_TYPES.PRIVACY_SHIP
+    ) {
+      return this.renderPrivacyForm();
+    }
+    const questionnaireText = GATE_QUESTIONNAIRES[this.gate.gate_type];
+    if (!questionnaireText) return html`No questions`;
+    const markup =
+      typeof questionnaireText == 'string'
+        ? autolink(questionnaireText)
+        : questionnaireText;
+    return html`
+      <!-- prettier-ignore -->
```

Comment:
What is the ignore needed here for?

Reply:
Gone.  I think I had that because in a previous iteration I was using `white-space: pre-line`.  All the textual questions are written as markup now rather than text with line-breaks, so it should not matter if prettier decides to break the next line into three.  And, prettier doesn't seem to want to do that now anyway.